quantity.hpp

```diff
@@ -63,6 +71,218 @@ using    day_scaling = std::ratio_multiply<  hour_scaling, scale<24>>;
+template<typename ID, std::intmax_t N, std::intmax_t D>
+struct mul_tag<tag<ID,N,D>,no_tag>                              {using type = tag<ID,N,D>;};
+template<typename ID, std::intmax_t N1, std::intmax_t D1
+                    , std::intmax_t N2, std::intmax_t D2>
+struct mul_tag<tag<ID,N1,D1>,tag<ID,N2,D2>>                     {using type = typename tag_add<tag<ID,N1,D1>,tag<ID,N2,D2>>::type;};
+
+
+template<typename Tag1, typename Tag2>
+struct div_tag;
+
+template<>
+struct div_tag<no_tag,no_tag>                                   {using type = no_tag;};
+template<typename ID, std::intmax_t N, std::intmax_t D>
+struct div_tag<no_tag,tag<ID,N,D>>                              {using type = tag<ID,N,D>;};
+template<typename ID, std::intmax_t N, std::intmax_t D>
+struct div_tag<tag<ID,N,D>,no_tag>                              {using type = tag<ID,N,D>;};
+template<typename ID, std::intmax_t N1, std::intmax_t D1
+                    , std::intmax_t N2, std::intmax_t D2>
+struct div_tag<tag<ID,N1,D1>,tag<ID,N2,D2>>                     {using type = typename tag_subtract<tag<ID,N1,D1>,tag<ID,N2,D2>>::type;};
+
+
+template<typename Tag>
+struct sqrt_tag : create_tag<typename Tag::id, std::ratio_divide<typename Tag::exponent,std::ratio<2>>::num
+                                             , std::ratio_divide<typename Tag::exponent,std::ratio<2>>::den> {};
+
+
+template<typename Tag, int Power>
+struct  pow_tag : create_tag<typename Tag::id, std::ratio_multiply<typename Tag::exponent,std::ratio<Power>>::num
+                                             , std::ratio_multiply<typename Tag::exponent,std::ratio<Power>>::den > {};
+
+
+template<typename Tag1, typename Tag2>
+struct are_tags_compatible;
+template< typename ID1, std::intmax_t N1, std::intmax_t D1
+        , typename ID2, std::intmax_t N2, std::intmax_t D2 >
+struct are_tags_compatible<tag<ID1,N1,D1>,tag<ID2,N2,D2>> : std::integral_constant<bool, tag<ID1,N1,D1>::is_no_tag::value
+	                                                                                  or tag<ID2,N2,D2>::is_no_tag::value
+	                                                                                  or detail::is_same_v<ID1,ID2> > {
+};
+template<typename Tag1, typename Tag2>
+struct are_tags_compatible : are_tags_compatible< typename create_tag<Tag1>::type
+	                                            , typename create_tag<Tag2>::type > {};
+
+}
+
+/**
+ * @brief Tag creation
+ *
+ * This creates a tag from an ID and a ratio.
+ *
+ * @tparam  ID  tag ID
+ * @tparam Num  tag ratio numerator
+ * @tparam Den  tag ratio denominator
+ *
+ * @note If you pass a tag as the ID parameter, Num and Dem will be ignored
+ *       and the tag returned instead. This allows this meta function to be
+ *       called with either a tag ID or a tag and always return a valid tag,
+ *       in turn allowing users to provide tag names only, rather than to
+ *       create tags each time.
```

Comment:
As noted above, as far as I understand, this will not create a new tag, just an alias for the existing provided tag.

I don't really understand the use case either; if I need a new tag, I will write e.g.
`using my_tag = create_tag_t<struct my_tag_struct>::type`

What is the intended meaning of then doing
`using my_other_tag = create_tag_t<my_tag>::type`

Reply:
This is used by the library to allow users to pass either tag IDs (which is what users are concerned with) or tags (which they get, for example, when they refer to a quantity type's tag type) where tags are expected. Running both through `create_tag_t` will turn the tag IDs into tags, and return tags "unharmed". I tried to explain this in the comment, but I guess I wasn't clear enough. 

The text now reads:

```
 * @note While users can invoke this meta-function to create tags from their
 *       tag IDs, they can also just pass tag IDs, which the library will then
 *       turn into tags.
 *       In order for the library to do this, this meta-function can be
 *       invoked with either a tag ID (and, optionally, the tag exponent's
 *       numerator and denominator), or with a tag. If a tag is passed as
 *       the ID parameter, Num and Dem will be ignored and the tag returned
 *       instead. This allows the library to pass whatever users provide
 *       through this meta function and get a valid tag from it.
```

HTH.